### PR TITLE
Clean up part file only once, not twice on error

### PR DESCRIPTION
**Pull request by PVince81**
Follow up PR for https://github.com/owncloud/core/pull/17189#issuecomment-117105185

Please review @nickvergessen @MorrisJobke @icewind1991

### CODE REVIEW

File: lib/private/connector/sabre/file.php [hunk 1]
```diff
@@ -352,6 +351,7 @@ private function createFileChunked($data) {
 			$needsPartFile = $this->needsPartFile($storage);
 
 			try {
+				$partFile = null;
```
Comment (nickvergessen) [inline lib/private/connector/sabre/file.php]: put this before the try please:
bildschirmfoto vom 2015-07-01 09 44 43

Comment (PVince81) [inline lib/private/connector/sabre/file.php]: I didn't aim high enough when shooting that line. Will do.

**Comment by PVince81** [comment]
Sleep well Jenkins... I force pushed.

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by nickvergessen** [comment]
:+1: and backport :see_no_evil:

**Comment by PVince81** [comment]
@karlitschek please approve backport to stable8.1

**Comment by PVince81** [comment]
Shit, I just found another issue, I'll add this to this commit.
:-1: for now

**Comment by PVince81** [comment]
if `$needsPartFile` is false, this code https://github.com/owncloud/core/blob/master/lib/private/connector/sabre/file.php#L149 would try and delete the original file... (can happen with s2s)

**Comment by PVince81** [comment]
We probably shouldn't have rushed https://github.com/owncloud/core/pull/17189... :see_no_evil:

**Comment by PVince81** [comment]
See https://github.com/owncloud/core/commit/cf245b80fe8bf0aed2b227e7f2cef5a15d3c5c8b

This will prevent deleting the real file if an upload failed for some reason through s2s (locking or other)

Please re-review.

**Comment by PVince81** [comment]
@owncloud-bot retest this please

**Comment by PVince81** [comment]
Jenkins PR: https://github.com/owncloud/core/pull/17281

**Comment by PVince81** [comment]
I raised a ticket for the part file with real file mix up: https://github.com/owncloud/core/issues/17284
It contains reproduction steps.

**Comment by karlitschek** [comment]
:+1:  please backport

**Comment by nickvergessen** [comment]
:+1: confirmed it fixes #17284 now as well.
(Missing "Fix #17284" from PR description? )

**Comment by PVince81** [comment]
The PR wasn't originally intended to fix #17284, I hijacked it to make it so :smile: 

Thanks for testing.

**Comment by PVince81** [comment]
Backport PR https://github.com/owncloud/core/pull/17289

**Pull request merged**